sdk/iothub/src/service/mod.rs

```diff
@@ -1,21 +1,35 @@
-use azure_core::errors::AzureError;
+use std::sync::Arc;
+
+use azure_core::{errors::AzureError, HttpClient};
 use base64::{decode, encode_config};
 use hmac::{Hmac, Mac, NewMac};
-use hyper::Method;
+use http::request::Builder as RequestBuilder;
+use http::{header, Method};
 use sha2::Sha256;
 
-pub mod directmethod;
-pub mod identity;
-pub mod twin;
-
-use crate::service::directmethod::DirectMethod;
-use crate::service::identity::{
-    delete_identity, get_identity, Device, DeviceIdentityBuilder, IdentityOperation, Module,
+/// The requests module contains any request that the IoT Hub service client can perform.
+pub mod requests;
+/// The resources module contains various types that some of the requests or responses use.
+pub mod resources;
+/// The response module contains responses for the requests that the IoT Hub service client can perform.
+pub mod responses;
+
+use crate::service::requests::{
+    CreateOrUpdateDeviceIdentityBuilder, CreateOrUpdateModuleIdentityBuilder,
+    DeleteIdentityBuilder, GetIdentityBuilder, GetTwinBuilder, InvokeMethodBuilder,
+    UpdateOrReplaceTwinBuilder,
+};
+use crate::service::resources::identity::IdentityOperation;
+use crate::service::responses::{
+    DeviceIdentityResponse, DeviceTwinResponse, ModuleIdentityResponse, ModuleTwinResponse,
 };
-use crate::service::twin::{get_twin, DesiredTwinBuilder, DeviceTwin, ModuleTwin};
 
+/// The API version to use for any requests
 pub const API_VERSION: &str = "2020-05-31-preview";
 
+/// A general error having to do with the IoTHub.
+pub type IoTHubError = Box<dyn std::error::Error + Sync + Send>;
```

Comment:
This is fine for now, but we should probably moved to more structured errors at some point.